fix: reset form on closing

shiv7shukla opened the pull request:
## Proposed Changes

Fixes #14046
- Moved reset logic out of the onSubmit wrapper to prevent RHF state corruption.
- Added a resetFormCompletely() helper to clear values, errors, and internal RHF state.
- Applied a dynamic key={resetCounter} to the element to force a full unmount/remount.

Tagging: @ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate the bug or test the new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is placed in I18n files.
- [ x ] Prepare a screenshot or demo video for the changelog entry and attach it to the issue.
- [ x ] Request peer reviews.
- [ ] Complete QA on mobile devices.
- [ ] Complete QA on desktop devices.
- [ ] Add or update Playwright tests for related changes

https://github.com/user-attachments/assets/7e53cbb7-bb79-49d4-af11-83195ee57eb2



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed form reset timing so fields are reliably cleared when opening resource category forms and when toggling edit mode.
  * Made reset behavior respond to editing state changes to avoid stale form values.

* **New Features**
  * Forms now remount or fully reset to ensure a fresh, consistent state when creating or editing resources.
  * Charge item definition forms fully reset after create/update and on cancel for predictable behavior.

* **Accessibility**
  * Added an accessible label to the drawer close control for improved screen reader support.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

shiv7shukla committed:
fix: reset form on closing

AdityaJ2305 reviewed (commented):
<img width="221" height="79" alt="Image" src="https://github.com/user-attachments/assets/b5f340a0-c336-4221-bbc1-19a4dba1bee1" />

@shiv7shukla Discard unwanted changes and fix the build failing

shiv7shukla committed:
chore: discard unwanted changes

shiv7shukla committed:
chore: implement changes by coderabbit

shiv7shukla committed:
chore: implement changes by coderabbit

shiv7shukla commented:
> <img alt="Image" width="221" height="79" src="https://private-user-images.githubusercontent.com/148325158/523366204-b5f340a0-c336-4221-bbc1-19a4dba1bee1.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjUwOTY3ODYsIm5iZiI6MTc2NTA5NjQ4NiwicGF0aCI6Ii8xNDgzMjUxNTgvNTIzMzY2MjA0LWI1ZjM0MGEwLWMzMzYtNDIyMS1iYmMxLTE5YTRkYmExYmVlMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIwN1QwODM0NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jYjc2YjBhNGE0OGM2NjIyNDZiYTE0OTEyMjMyZDkxOTI5YjlmMDMzNTRlMTk4YjQxNjM0OWYzMTEzNmYzOWFkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.1kNwSAikS_zbNqU6Ysy-XvDuC2WXyv0NJT9D46BzWBw">
> @shiv7shukla Discard unwanted changes and fix the build failing

@AdityaJ2305 I have removed the unwanted changes and build is also fixed.
<img width="1871" height="851" alt="Screenshot 2025-12-07 140812" src="https://github.com/user-attachments/assets/5eaff6ec-70e2-4a7e-a8e3-836dada8b754" />

shiv7shukla committed:
Merge branch 'develop' into issues/14046/form-reset-errors
